### PR TITLE
e2e tests: Always scrollIntoView in enterTestCreditCardDetails

### PR DESCRIPTION
**Pull request by sirbrillig**
#### Changes proposed in this Pull Request

Some e2e tests are failing (with the cryptic error `TimeoutError: Timed out waiting for element with undefined of 'undefined' to be clickable`) which is thrown by `clickWhenClickable` [if the selector does not exist](https://github.com/Automattic/wp-calypso/blob/08a46585106b1a4981b166217372b68b99358d88/desktop/e2e/tests/lib/driver-helper.js#L39). This is because the checkout cardholder name field is slightly too high on the page and cannot be selected. The test code in `enterTestCreditCardDetails` already accounts for this on mobile by scrolling, but on desktop it did not. This PR changes that behavior so it is always scrolled into view.

#### Testing instructions

If you're set up for running e2e tests, then do:

`env BROWSERSIZE=desktop ./node_modules/.bin/mocha --grep "Sign up for a domain only purchase coming in from wordpress.com/domains in EUR currency" specs/wp-signup-spec.js`

You'll still see the tests fail, but they'll fail because of [the sidebar missing](https://github.com/Automattic/wp-calypso/issues/46684) after the purchase completes (fixed in [a separate PR](https://github.com/Automattic/wp-calypso/pull/46711)), not before the purchase.

### CODE REVIEW

**Comment by sirbrillig** [comment]
Hm... I can see that there's still a similar failing test in the `wp-reader-spec.js`. I wonder if something changed with selenium.

There's also some issues selecting the sidebar. I don't think that's related to this PR.

File: desktop/e2e/tests/lib/driver-helper.js [hunk 1]
```diff
@@ -291,7 +297,7 @@ exports.selectElementByText = async function ( driver, selector, text ) {
 			async ( e ) => ( await e.getText() ) === text
 		);
 	};
-	return await this.clickWhenClickable( driver, element );
+	return await this.clickWhenClickable( driver, element, `while looking for '${ text }'` );
```
Comment (michaeldcain) [inline desktop/e2e/tests/lib/driver-helper.js:300]: Nice.

**Review by michaeldcain — APPROVED** [review]
Thanks for fixing these!

File: desktop/e2e/tests/lib/driver-helper.js (hunk 1 above)
Comment (sirbrillig) [reply desktop/e2e/tests/lib/driver-helper.js:300]: Yeah, this should help diagnosing future issues like this. It took me a while to figure out where that error was coming from.

**Pull request merged**